tests/test_attack_report_formatter.py

```diff
@@ -209,13 +342,18 @@ def test_auc_pvals_module(self):
 
         self.assertIn("Summary of AUC p-values", str(f))
 
+        _ = json_formatted["WorstCaseAttack"].pop("attack_experiment_logger")
+        f = SummariseAUCPvalsModule(json_formatted)
+
     def test_fdif_pvals_module(self):
         """test the SummariseFDIFPvalsModule"""
         json_formatted = self.get_test_report()
         f = SummariseFDIFPvalsModule(json_formatted)
         _ = str(f)
         _ = f.process_dict()
-        _ = f.get_metric_list(json_formatted["attack_experiment_logger"])
+        _ = f.get_metric_list(
+            json_formatted["WorstCaseAttack"]["attack_experiment_logger"]
+        )
 
         self.assertIn("Summary of FDIF p-values", str(f))
 
```

Comment:
Though I can see that you have updated the tests regarding the use of classes in attack_report_formatter module; however, I wonder if would it not be helpful to update the examples to demonstrate this functionality. 

I can see our examples are already cluttered in a way we are trying to demonstrate all possible scenarios, and we need to think about how can we make them clean. I believe this may be done by creating separate files to demonstrate running attacks from the command line and using programmatically for each type of attack.